src/store/tools/useAnnotationTool.ts

```diff
@@ -54,10 +48,10 @@ export const useAnnotationTool = <Tool extends AnnotationTool>({
     return TOOL_COLORS[colorIndex];
   }
 
-  const idGenerator = useIDGenerator();
+  const idStore = useIDStore();
 
   const addTool = (tool: ToolPatch): ToolID => {
-    const id = idGenerator.nextID() as ToolID;
+    const id = idStore.nextID() as ToolID;
```

Comment:
If you want to access `this.$id` in `addTool`, you'd need to change this to be the following. It works because `addTool` gets exposed as a store method, and pinia will go through and bind `this` to each exposed store method.

```
function addTool(this: _This, tool: ToolPatch): ToolID {
  const id = this.$id.nextID() as ToolID;
```